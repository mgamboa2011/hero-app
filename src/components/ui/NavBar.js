import React, { useContext } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types'

export const Navbar = () => {

    const { user: { name }, dispatch } = useContext(AuthContext)
    const navigate = useNavigate();
    const handleLogout = () => {
        navigate('/login', { replace: true });
        dispatch({
            type: types.logout
        })
    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">

                <Link className="navbar-brand" to="/">Asociaciones</Link>

                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <Link className="nav-link" to="/marvel">Marvel</Link>
                        <Link className="nav-link" to="/dc">DC</Link>
                        <Link className="nav-link" to="/search">Search</Link>
                    </div>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                <ul className="navbar-nav ms-auto">
                    <span className="nav-item nav-link text-info">
                        {name}
                    </span>
                    <button
                        className="nav-link nav-item btn"
                        onClick={handleLogout}
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}