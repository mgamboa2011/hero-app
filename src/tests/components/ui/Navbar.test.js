import { Navbar } from "../../../components/ui/NavBar"
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/AuthContext";
import { types } from "../../../types/types";

describe('Prueba en <Navbar/>', () => {
    const navigateMock = {
        push: jest.fn(),
        location: {},
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Pepe'
        }
    };
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter navigate={navigateMock}>
                <Navbar />
            </MemoryRouter>
        </AuthContext.Provider>
    )
    afterEach(() => {
        jest.clearAllMocks();
    });

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');
    });

    test('debe llamar el logout y el navigate', () => {
        wrapper.find('button').simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
        //expect(navigateMock.replace).toHaveBeenCalledWith('/login');
    })


})
