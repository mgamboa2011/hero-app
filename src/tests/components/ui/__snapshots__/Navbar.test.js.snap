// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <Navbar/> debe mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ms-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pepe
      </span>
      <button
        className="nav-link nav-item btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
